### PR TITLE
diawi-upload 0.0.5

### PR DESCRIPTION
**Pull request by deanWombourne**
TagCheck

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [ ? ] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

* Step is a wrapper around bitrise node library - it can't really be tested without a valid upload key and a test .ipa file. There is a test file in the step's repo, but the upload key requires a Diawi account. It's an open issue for the test to simply check the format of the command run without actually running it.

### CODE REVIEW

File: steps/diawi-upload/0.0.5/step.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      It's free to sign up and get one, just go here: https://dashboard.diawi.com/signup.
+      You create an api token in the profile tab.
+    is_expand: true
+    is_required: true
+    summary: Diawi API token
+    title: API token
+- filename: $BITRISE_IPA_PATH
+  opts:
+    description: |
+      The path to the file to upload - defaults to an IPA created by a bitrise archive step
```
Comment (koral--) [inline steps/diawi-upload/0.0.5/step.yml]: What if there is no IPA produced but APK? Or multiple APKs for various flavors etc?
Maybe this input should accept array of file paths and also output array of urls?

File: steps/diawi-upload/0.0.5/step.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+inputs:
+- api_token: $DIAWI_TOKEN
+  opts:
+    description: |
+      The API token for Diawi.
+
+      It's free to sign up and get one, just go here: https://dashboard.diawi.com/signup.
+      You create an api token in the profile tab.
+    is_expand: true
+    is_required: true
```
Comment (koral--) [inline steps/diawi-upload/0.0.5/step.yml]: It should be also sensitive probably.

File: steps/diawi-upload/0.0.5/step.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  brew:
+  - name: npm
+  apt_get:
+  - name: npm
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: false
+run_if: ""
+inputs:
+- api_token: $DIAWI_TOKEN
```
Comment (koral--) [inline steps/diawi-upload/0.0.5/step.yml]: In general default value should only be set to variables which can be exported by previous steps (or Bitrise CLI or bitrise.io) not to custom ones.

File: steps/diawi-upload/0.0.5/step.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- opts:
+    description: Users will have to provide this to download the app from Diawi. No
+      value in here will allow anyone to download the app.
+    is_expand: true
+    is_required: false
+    is_sensitive: true
+    summary: Users will have to provide this to download the app from Diawi. No value
+      in here will allow anyone to download the app.
+    title: Password
+  password: $DIAWI_UPLOAD_PASSWORD
```
Comment (godrei) [inline steps/diawi-upload/0.0.5/step.yml]: Same as here: https://github.com/bitrise-io/bitrise-steplib/pull/2395/files#r380228134

Another issue with this kind of default values, is that the Workflow Editor will not highlight this input as required but not yet set.

**Comment by godrei** [comment]
Hi @deanWombourne ,
thanks for the contribution.

Please see the comments and fix them.
For this comment: https://github.com/bitrise-io/bitrise-steplib/pull/2395#discussion_r380226539, I think the easiest solution is to remove the default value and let users fill it, based on there project type and steps in the given workflow.

@koral-- thanks for the review!

File: steps/diawi-upload/0.0.5/step.yml (hunk 1 above)
Comment (deanWombourne) [reply steps/diawi-upload/0.0.5/step.yml]: Actually, I do no validation of the input file - I leave that to Diawi - so passing in an apk will Just Work.

I've created an issue for passing in multiple files - https://github.com/deanWombourne/bitrise-step-diawi-upload/issues/10

**Comment by deanWombourne** [comment]
OK @godrei @koral-- I've removed the default parameter values, but resharing has created a new pr here: https://github.com/bitrise-io/bitrise-steplib/pull/2398

**Pull request closed without merging**